sql/mdl.h

```diff
@@ -1102,7 +1102,7 @@ class MDL_context
   /**
     Ticket hash. Stores only locked tickets.
   */
-  open_address_hash<MDL_key_trait<MDL_ticket>, hash_trait<MDL_ticket> > ticket_hash;
+  open_address_hash<MDL_key_trait<MDL_ticket>, hash_trait<MDL_ticket, MDL_key_trait<MDL_ticket> > > ticket_hash;
```

Comment:
you see, this way it makes the declaration harder, which better should be avoided. Actually, the only thing we're concerned is the return type of `get_key` the rest can be cared out inside the hash code.